src/firefly/js/core/LayoutCntlr.js

```diff
@@ -314,10 +316,28 @@ export function dropDownHandler(layoutInfo, action) {
     switch (action.type) {
         case CHART_ADD:
         case TBL_RESULTS_ADDED:
+        case TBL_RESULTS_ACTIVE:
+        case TABLE_LOADED:
+            const tbl = action.type === CHART_ADD ? getTblById(action.payload.groupId) : getTblById(action.payload.tbl_id);
+            if (action.payload.options?.tbl_group !== 'main') {
+                return layoutInfo; //do not load results view if tbl group is not main
+            }
+            //Todo: const isUploadTbl= getBooleanMetaEntry(tbl, MetaConst.UPLOAD_TABLE, false);
+            if (tbl?.request?.META_INFO?.UploadTable) {
+                return layoutInfo;
+            }
+            return smartMerge(layoutInfo, {dropDown: {visible: false}});
         case REPLACE_VIEWER_ITEMS :
         case ImagePlotCntlr.PLOT_IMAGE :
             return smartMerge(layoutInfo, {dropDown: {visible: false}});
         case ImagePlotCntlr.PLOT_IMAGE_START :
+            const VISUALIZED_TABLE_IDS = action.payload.attributes.VISUALIZED_TABLE_IDS;
+            if (VISUALIZED_TABLE_IDS && VISUALIZED_TABLE_IDS.length > 0) {
+                const lastId = VISUALIZED_TABLE_IDS[VISUALIZED_TABLE_IDS.length - 1];
+                // Check if the last entry contains 'Upload_Tbl' - and return layoutInfo as is if it does
+                const containsUploadTbl = lastId.includes('Upload_Tbl');
+                if (containsUploadTbl) return layoutInfo;
+            }
             const {useForSearchResults= true}= action.payload.pvOptions;
             const visible= !useForSearchResults && layoutInfo.dropDown.visible;
```

Comment:
This will not work in all cases but, these two lines are a slight modification of the existing code and fix an existing problem.
```js
const {useForSearchResults= true, useForCoverage}= action.payload.pvOptions;
const visible= (!useForCoverage && !useForSearchResults) && layoutInfo.dropDown.visible;
```